tests/streamlit_ui_tests/set_column_formula_tests.spec.ts

```diff
@@ -0,0 +1,288 @@
+    expect(cellValues).toEqual(['1', '4', '7', '10']);
+});
+
+test('Reference column by clicking on column header', async ({ page }) => {
+    const columnHeader = 'Column4';
+
+    const mito = await getMitoFrameWithTestCSV(page);
+    await createNewColumn(page, mito, 3, columnHeader);
+
+    const cell = await getCellAtRowIndexAndColumnName(mito, 0, columnHeader);
+    await cell.dblclick();
+    await mito.getByRole('textbox').fill('=SUM(');
+
+    await (mito.getByText('Column1')).click();
+    // Type the rest of the formula
+    await (mito.getByRole('textbox')).press(')')
+    await mito.locator('#cell-editor-input').press('Enter');
+    await awaitResponse(page);
+
+    const cellValues = await getValuesInColumn(mito, columnHeader);
+    expect(cellValues).toEqual(['22', '22', '22', '22']);
+});
+
+test('Reference cell in previous row by clicking on it', async ({ page }) => {
+    const columnHeader = 'Column4';
+
+    const mito = await getMitoFrameWithTestCSV(page);
+    await createNewColumn(page, mito, 3, columnHeader);
+
+    const cell = await getCellAtRowIndexAndColumnName(mito, 1, columnHeader);
+    await cell.dblclick();
+
+    await (mito.locator('.mito-grid-cell').first()).click();
+    await mito.locator('#cell-editor-input').press('Enter');
+    await awaitResponse(page);
+
+    const cellValues = await getValuesInColumn(mito, columnHeader);
+    expect(cellValues).toEqual(['0', '1', '4', '7']);
+});
+
+test('Use arrow keys to select cell', async ({ page }) => {
+    const columnHeader = 'Column4';
+
+    const mito = await getMitoFrameWithTestCSV(page);
+    await createNewColumn(page, mito, 3, columnHeader);
+
+    const cell = await getCellAtRowIndexAndColumnName(mito, 0, columnHeader);
+    await cell.dblclick();
+    await mito.getByRole('textbox').fill('=');
+
+    await page.keyboard.press('ArrowLeft');
+
+    await mito.locator('#cell-editor-input').press('Enter');
+    await awaitResponse(page);
+
+    const cellValues = await getValuesInColumn(mito, columnHeader);
+    expect(cellValues).toEqual(['3', '6', '9', '12']);
+});
+
+test('Use arrow keys to select rolling range', async ({ page }) => {
+    const columnHeader = 'Column4';
+
+    const mito = await getMitoFrameWithTestCSV(page);
+    await createNewColumn(page, mito, 3, columnHeader);
+
+    const cell = await getCellAtRowIndexAndColumnName(mito, 0, columnHeader);
+    await cell.dblclick();
+    await mito.getByRole('textbox').fill('=SUM(');
+
+    await page.keyboard.press('ArrowLeft');
+    await page.keyboard.press('Shift+ArrowLeft');
+    await page.keyboard.press(')');
+
+    await mito.locator('#cell-editor-input').press('Enter');
+    await awaitResponse(page);
+
+    const cellValues = await getValuesInColumn(mito, columnHeader);
+    expect(cellValues).toEqual(['5', '11', '17', '23']);
+});
+
+
+test('No formula around rolling range displays [object Object]', async ({ page }) => {
```

Comment:
So this is something that we want? Seems not super intuitive

Reply:
ehh, yeah not sure if we actually do want it. But I guess we want to know if we change something that breaks it. I should probably leave that as a comment in this test so we have context if this test ever fails. Good point!

Reply:
Ok yeah just making sure!